Modules/Microsoft.PowerShell.UnixCompleters/Microsoft.PowerShell.UnixCompleters/BashUtilCompleter.cs

```diff
@@ -0,0 +1,309 @@
+                COMP_POINT: cursorPosition,
+                completerFunction,
+                wordToComplete,
+                previousWord);
+
+            List<string> completionResults = InvokeBashWithArguments(completionCommand)
+                .Split('\n')
+                .Distinct(StringComparer.Ordinal)
+                .ToList();
+
+            completionResults.Sort(StringComparer.Ordinal);
+
+            string previousCompletion = null;
+            foreach (string completionResult in completionResults)
+            {
+                if (string.IsNullOrEmpty(completionResult))
+                {
+                    continue;
+                }
+
+                int equalsIndex = wordToComplete.IndexOf('=');
+
+                string completionText;
+                string listItemText;
+                if (equalsIndex >= 0)
+                {
+                    completionText = wordToComplete.Substring(0, equalsIndex) + completionResult;
+                    listItemText = completionResult;
+                }
+                else
+                {
+                    completionText = completionResult;
+                    listItemText = completionText;
+                }
+
+                if (completionText.Equals(previousCompletion))
+                {
+                    listItemText += " ";
+                }
+
+                previousCompletion = completionText;
+
+                yield return new CompletionResult(
+                    completionText,
+                    listItemText,
+                    CompletionResultType.ParameterName,
+                    completionText);
+            }
+        }
+
+        private string ResolveCommandCompleterFunction(string commandName)
+        {
+            if (string.IsNullOrEmpty(commandName))
+            {
+                throw new ArgumentException(nameof(commandName));
+            }
+
+            string completerFunction;
+            if (_commandCompletionFunctions.TryGetValue(commandName, out completerFunction))
+            {
+                return completerFunction;
+            }
+
+            string resolveCompleterInvocation = string.Format(s_resolveCompleterCommandTemplate, commandName);
+            completerFunction = InvokeBashWithArguments(resolveCompleterInvocation).Trim();
+            _commandCompletionFunctions[commandName] = completerFunction;
+
+            return completerFunction;
+        }
+
+        private string InvokeBashWithArguments(string argumentString)
+        {
+            using (var bashProc = new Process())
+            {
+                bashProc.StartInfo.FileName = this._bashPath;
+                bashProc.StartInfo.Arguments = argumentString;
+                bashProc.StartInfo.UseShellExecute = false;
+                bashProc.StartInfo.RedirectStandardOutput = true;
+                bashProc.Start();
+
+                return bashProc.StandardOutput.ReadToEnd();
+            }
```

Comment:
I'm not sure I understand

Reply:
Sorry:
```c#
            using var bashProc = new Process());
            bashProc.StartInfo.FileName = this._bashPath;
            bashProc.StartInfo.Arguments = argumentString;
            bashProc.StartInfo.UseShellExecute = false;
            bashProc.StartInfo.RedirectStandardOutput = true;

            bashProc.Start();
            return bashProc.StandardOutput.ReadToEnd();
```

Reply:
Oh! Yeah, I've tried that on a few projects, but even though VS suggests it, it's still in a C# preview version. So to prevent needing to set preview flags in the csproj, I just use the traditional syntax